internal/gensample/gensample.go

```diff
@@ -0,0 +1,740 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gensample
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base32"
+	"encoding/json"
+	"fmt"
+	"go/format"
+	"log"
+	"os"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/googleapis/gapic-generator-go/internal/errors"
+	"github.com/googleapis/gapic-generator-go/internal/gensample/schema_v1p2"
+	"github.com/googleapis/gapic-generator-go/internal/license"
+	"github.com/googleapis/gapic-generator-go/internal/pbinfo"
+	"github.com/googleapis/gapic-generator-go/internal/printer"
+	yaml "gopkg.in/yaml.v2"
+)
+
+const expectedSampleConfigType = "com.google.api.codegen.samplegen.v1p2.SampleConfigProto"
+const expectedSampleConfigVersion = "1.2.0"
+
+// InitGen creates a new sample generator.
+func InitGen(desc []*descriptor.FileDescriptorProto, sampleFnames []string, gapicFname string, clientPkg string, nofmt bool) (*generator, error) {
+
+	gen := generator{
+		imports:  map[pbinfo.ImportSpec]bool{},
+		desc:     desc,
+		descInfo: pbinfo.Of(desc),
+	}
+
+	if p := strings.IndexByte(clientPkg, ';'); p >= 0 {
+		gen.clientPkg = pbinfo.ImportSpec{Path: (clientPkg)[:p], Name: (clientPkg)[p+1:]}
+	} else {
+		return nil, errors.E(nil, "need -clientPkg in 'url/to/client/pkg;name' format, got %q", clientPkg)
+	}
+
+	var wg sync.WaitGroup
+	errChan := make(chan error)
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		if gapicFname != "" {
+			f, err := os.Open(gapicFname)
+			if err != nil {
+				errChan <- errors.E(err, "cannot read GAPIC config file: %q", gapicFname)
+				return
+			}
+			defer f.Close()
+
+			if err := yaml.NewDecoder(f).Decode(&gen.gapic); err != nil {
+				errChan <- errors.E(err, "error reading GAPIC config file: %q", gapicFname)
+				return
+			}
+		}
+		errChan <- nil
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		if err := gen.readSampleConfigFiles(sampleFnames); err != nil {
+			errChan <- err
+			return
+		}
+		errChan <- nil
```

Comment:
this could probably just be 

```go
defer wg.Done()
errChan <- gen.readSampleConfigFiles(sampleFnames)
```

Or save the return value into `err` and send that to the `chan` if you'd prefer.